### ISSUE TITLE
Units in map when using 3857 projection

### ISSUE THREAD

**Issue by gustavovelascoh** [MatthewDaws/TileMapBase#24]
Hi, as far as I understand when using the default Extent class the units in the plotted map are the values in the 0-1 range from the conversion from latitude, longitude to x,y  in a unit square.

However, when I use extent.to_project_3857() the units are in the order of 10^3 - 10^4 (as in [the example notebook](https://github.com/MatthewDaws/TileMapBase/blob/master/notebooks/Example.ipynb)).

What units are those? I thought they were meters but after comparing it with the measurement tool in Google maps, there is an error of about 66%.

**Comment by MatthewDaws** [comment]
It _should be_ conforming to the standard [EPSG:3857](https://spatialreference.org/ref/sr-org/6864/)  It has been some years now since I thought hard about projections, but if I understand e.g. the "well known text" format for 3857, then the units should be meters.

If you can confirm that `extent.to_project_3857()` is not agreeing with how another tool projects to EPSG:3857 then I will investigate further.

**Comment by gustavovelascoh** [comment]
Thanks for your quick reply. I'll detail my case:

I'm working around this location 52.0705818, -9.5490888 ([GoogleMaps](https://www.google.com/maps/@52.0702657,-9.5494995,335m/data=!3m1!1e3), [OpenStreetMap](https://www.openstreetmap.org/#map=18/52.07028/-9.54982)). As can be seen, there are two buildings and from google maps measurement tool, one has ~100 m side length and the other ~ 60m.

![Screenshot from 2021-05-27 12-32-21](https://user-images.githubusercontent.com/2431850/119820428-5f45cf00-bee9-11eb-869b-ee57416af2ee.png)
![Screenshot from 2021-05-27 12-32-58](https://user-images.githubusercontent.com/2431850/119820444-61a82900-bee9-11eb-8609-cc8562ec6dc4.png)

Here below are the plots using both the default `extent` and the `.to_project_3857()` option. From the first one, the units are reasonable as they are part of a unit square covering almost the whole earth. But in the second plot, the building with a 60m side, is about 100 units on that scale (from 820 to 920)  and the supposed 100m side is more than 150 units (1062850 to 1063000).

![extent_default](https://user-images.githubusercontent.com/2431850/119820584-87353280-bee9-11eb-84ca-06518754b419.png)
![extent_3857](https://user-images.githubusercontent.com/2431850/119820592-88fef600-bee9-11eb-99be-f88e0a60dd49.png)

Then, I made a custom projection using a defined UTM zone (29U north in my case) and got the plot below. In this one, the units are closer to the measured values. 60 (-10 to 50) for the left side building and  ~100 (462360 to 462460) for the bottom building.
![extent_UTM](https://user-images.githubusercontent.com/2431850/119821438-7933e180-beea-11eb-92dc-8a2a5d3308d1.png)

**Comment by MatthewDaws** [comment]
So, I think this is that EPSG:3857 doesn't do what you expect.  Here's some usage of `pyproj` which I ran in an IPython session:

      In [1]: import pyproj
      
      In [2]: p3857 = pyproj.CRS("EPSG:3857")
      
      In [3]: p4326 = pyproj.CRS("EPSG:4326")
      
      In [4]: tr = pyproj.Transformer.from_crs(p4326, p3857).transform
      
      In [5]: tr(52.0701843, -9.5490918)
      Out[5]: (-1063000.0367142241, 6812825.637814251)
      
      In [6]: tr(52.0704447, -9.5477270)
      Out[6]: (-1062848.1078731895, 6812872.795583437)
      
      In [7]: point1 = tr(52.0701843, -9.5490918)
      
      In [8]: point2 = tr(52.0704447, -9.5477270)
      
      In [9]: ( (point1[0]-point2[0])**2 + (point1[1]-point2[1])**2 ) ** 0.5
      Out[9]: 159.07931334014333

Here [EPSG:4326](https://spatialreference.org/ref/epsg/wgs-84/) is just raw longitude / latitude.  The coordinates I grabbed from [OpenStreetMap](https://www.openstreetmap.org/way/525487878) as the top right and left corners of the building.  End result: the distance between these points is 159 units.

**Comment by gustavovelascoh** [comment]
Thanks for the code. I did the same with the coordinates you provided using the projection I used and got 97.93 units (meters)

![image](https://user-images.githubusercontent.com/2431850/119833024-a6868c80-bef6-11eb-885c-ed7b0b619d69.png)

Also, I found this remark on the [EPSG:3857](https://epsg.io/3857) entry:

> Remarks: Uses spherical development of ellipsoidal coordinates. Relative to WGS 84 / World Mercator (CRS code 3395) errors of 0.7 percent in scale and differences in northing of up to 43km in the map (equivalent to 21km on the ground) may arise.

So, if I understood well, the error was expected (~62% in the example coordinates) when using EPSG:3857, but if a more precise scale is needed the approach I used is fine. Am I right on this?

Thanks again for your support.
